examples/synthesis_json.py

```diff
@@ -0,0 +1,159 @@
+# ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED
+# WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+# DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE FOR ANY
+# DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES
+# (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+# LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND
+# ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+# (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
+# SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+
+'''Extract the optimal distributions for the following features of the population of neurons:
+   soma: radius
+   basal dendrites: n_neurites
+   apical dendrites: n_neurites
+   axons: n_neurites
+   '''
+
+from neurom import ezy
+from neurom import stats
+import numpy as np
+import argparse
+import json
+from collections import OrderedDict
+import os
+
+
+def parse_args():
+    '''Parse command line arguments'''
+    parser = argparse.ArgumentParser(
+        description='Morphology fit distribution extractor',
+        epilog='Note: Outputs json of the optimal distribution \
+                and corresponding parameters.')
+
+    parser.add_argument('datapath',
+                        help='Path to morphology data file or directory')
+
+    return parser.parse_args()
+
+
+def extract_data(data_path, feature, params=None):
+    '''Loads a list of neurons, extracts feature
+       and transforms the fitted distribution in the correct format.
+       Returns the optimal distribution, corresponding parameters,
+       minimun and maximum values.
+    '''
+    population = ezy.load_neurons(data_path)
+
+    if params is None:
+        params = {}
+
+    feature_data = [getattr(n, 'get_' + feature)(*params) for n in population]
+
+    results = {}
+
+    try:
+        opt_fit = stats.optimal_distribution(feature_data)
+    except ValueError:
+        from itertools import chain
+        feature_data = list(chain(*feature_data))
+        opt_fit = stats.optimal_distribution(feature_data)
+
+    results["type"] = opt_fit[0]
+    results["params"] = opt_fit[1][0]
+
+    return results
+
+
+def transform_distribution(data, datamin=None, datamax=None):
+    '''Transform optimal distribution data into correct format
+       based on the distribution type.
+    '''
+    data_dict = OrderedDict()
+
+    if data["type"] == 'norm':
+        data_dict.update({"type": "normal"})
+        data_dict.update({"mu": data["params"][0]})
+        data_dict.update({"sigma": data["params"][1]})
+
+    elif data["type"] == 'expon':
+        data_dict.update({"type": "exponential"})
+        data_dict.update({"lambda": 1. / data["params"][1]})
+
+    elif data["type"] == 'uniform':
+        data_dict.update({"type": "uniform"})
+
+    if datamin is not None:
```

Comment:
Yes, I'm working on this now. However, I don't think that we should have the same name for the uniform parameters and the min max that are defined by the user. To avoid confusion, I will use [lower bound, upper bound] for now for the uniform and we can discuss this later. Is that ok for you?